src/Core/Services/IUserService.cs

```diff
@@ -65,6 +65,8 @@ public interface IUserService
     Task<bool> CheckPasswordAsync(User user, string password);
     Task<bool> CanAccessPremium(ITwoFactorProvidersUser user);
     Task<bool> HasPremiumFromOrganization(ITwoFactorProvidersUser user);
+    Task<IEnumerable<(Guid userId, bool twoFactorIsEnabled)>> TwoFactorIsEnabledAsync(IEnumerable<Guid> userIds);
+    Task<IEnumerable<(T user, bool twoFactorIsEnabled)>> TwoFactorIsEnabledAsync<T>(IEnumerable<T> users) where T : ITwoFactorProvidersUser;
```

Comment:
I used a generic type here because it allows the method to return the actual type that was passed in as the input. This means that after the method is called, you still have access to all the properties and methods of the original type, not just those defined in the `ITwoFactorProvidersUser` interface. If we only used `ITwoFactorProvidersUser`, we'd be limited to interacting with the properties and methods defined by that interface.